packages/web/components/pages/asset-info-page/token-details.tsx

```diff
@@ -177,59 +173,175 @@ const _TokenDetails = ({ className }: TokenDetailsProps) => {
 
 export const TokenDetails = observer(_TokenDetails);
 
-const TokenStats = observer(() => {
+const formatCompact = (value: PricePretty | number) => {
+  return formatPretty(value instanceof PricePretty ? value : new Dec(value), {
+    maximumSignificantDigits: 3,
+    notation: "compact",
+    compactDisplay: "short",
+    scientificMagnitudeThreshold: 30,
+  });
+};
+
+export const TokenStats = observer(() => {
   const { t } = useTranslation();
+  const [isExpanded, setIsExpanded] = useState(false);
 
-  const { coingeckoCoin, isLoadingCoingeckoCoin } = useAssetInfo();
+  const {
+    coinGeckoId,
+    coingeckoCoin,
+    isLoadingCoingeckoCoin = true,
+    tokenDenom,
+  } = useAssetInfo();
+
+  const { data: tokenMarket, isLoading: isLoadingTokenMarket } =
+    api.edge.assets.getUserMarketAsset.useQuery(
```

Comment:
Let's create a `getMarketAsset` that doesn't include the support for user address to simplify things a bit. It would be oriented around asset market data